tools/mpy-tool.py

```diff
@@ -1765,6 +1765,55 @@ def copy_section(file, offset, offset2):
             f.write(merged_mpy)
 
 
+def parse_extract_segments_arg(arg):
+    kinds = set()
+    if arg is not None:
+        for kind in arg.lower().split(','):
```

Comment:
If you use `arg.upper().split(","):` then it could be simply:
```py
try:
    kinds.add(getattr(MPYSegment, kind))
except AttributeError:
    raise Exception("unknown kind")
```
Then, this function could be written inline in the one below.  (A little simpler to have everything self contained in one function, IMO.)

Reply:
Oh, I'm used to use `lower` to work around internationalisation issues in input (f.ex the Turkish I problem).

I agree this is probably not really needed here, but old habits tend to die hard don't they.